src/main/java/com/inventoryservice/controller/CategoryController.java

```diff
@@ -0,0 +1,226 @@
+    @PutMapping("/{categoryId}")
+    @InventoryManagerAccess
+    public ResponseEntity<?> updateCategory(
+            @PathVariable Integer categoryId,
+            @RequestBody Category category
+    ) throws CategoryNotFoundException, CategoryArchivedException {
+        Category categories = categoryService.findCategoryById(categoryId);
+        if (categories == null) {
+            throw new CategoryNotFoundException("Category not found");
+        }
+        if (categories.isArchived()) {
+            throw new CategoryArchivedException("Category is archived and cannot be updated");
+        }
+        Category updatedCategory = categoryService.updateCategory(categoryId, category);
+        CategoryResponse categoryResponse = CategoryResponse.fromCategory(updatedCategory);
+        return new ResponseEntity<>(categoryResponse, HttpStatus.FOUND);
+    }
+
+    /*  To Add new SubCategory */
+    @PostMapping("/{categoryId}/subCategory")
+    @InventoryManagerAccess
+    public ResponseEntity<List<SubCategoryResponse>> addSubCategories(
+            @PathVariable("categoryId") Integer categoryId,
+            @RequestBody List<SubCategoryRequest> subCategoryRequests) {
+        Category category = categoryService.findCategoryById(categoryId);
+        if (category == null) {
+            throw new CategoryNotFoundException("Category not found");
+        }
+        List<SubCategory> subCategories = subCategoryRequests.stream()
+                .map(subCategoryRequest -> {
+                    SubCategory subCategory = SubCategoryRequest.toSubCategory(subCategoryRequest);
+                    subCategory.setCategory(category);
+                    System.out.println(subCategory);
+                    return subCategory;
+                })
+                .collect(Collectors.toList());
+        List<SubCategory> savedSubCategories = subcategoryService.saveSubcategories(subCategories);
+        List<SubCategoryResponse> subCategoryResponses = savedSubCategories.stream()
+                .map(SubCategoryResponse::fromSubcategory)
+                .collect(Collectors.toList());
+        return ResponseEntity.status(HttpStatus.CREATED).body(subCategoryResponses);
+    }
+
+    /*
+     * To get list of all SubCategory
+     * @return      list of all SubCategory .
+     */
+    @GetMapping(("/subCategory"))
+    public ResponseEntity<List<SubCategoryResponse>> getAllSubCategory() {
+        List<SubCategory> subCategories = subcategoryService.getAllSubCategory();
+        List<SubCategoryResponse> subCategoryResponse = subCategories.stream()
+                .map(SubCategoryResponse::fromSubcategory)
+                .collect(Collectors.toList());
+        return new ResponseEntity<>(subCategoryResponse, HttpStatus.OK);
+
+    }
+    /*
+      The method used to  delete the SubCategory.
+      This method  also check SubCategory is present or not   .
+      */
+    @DeleteMapping("/{categoryId}/subCategory/{subCategoryId}")
```

Comment:
> Lets have api which will aloow category/subcategory to archive . And to delete categorysubcategory we first check if it is archive then only it can be deleted.

created api   for category/subcategory to archive and un-archive it.